web/modules/custom/dpl_update/dpl_update.install

```diff
@@ -11,6 +11,93 @@
 
 use Drupal\Core\Extension\ModuleInstallerInterface;
 use Drupal\drupal_typed\DrupalTyped;
+use Drupal\recurring_events\Entity\EventInstance;
+use Drupal\recurring_events\Entity\EventSeries;
+
+/**
+ * Helper function to install modules.
+ *
+ * @param string[] $modules
+ *   The modules to install.
+ *
+ * @return string
+ *   The feedback message.
+ */
+function _dpl_update_install_modules(array $modules): string {
+  DrupalTyped::service(ModuleInstallerInterface::class, 'module_installer')->install($modules);
+  $modules_string = implode(', ', $modules);
+  return "Enabled modules: {$modules_string}.";
+}
+
+/**
+ * Linking new field inheritances with existing eventinstances.
+ *
+ * There is a fault in field_inheritance, when you add new fields/inheritances,
+ * it doesn't get updated on the old eventinstances until they get saved from
+ * the form.
+ * This is because the logic that links eventinstances and eventseries together
+ * is set by field_inheritance directly in the form_alter and form_submit.
+ * This helper function allows you to pass along a name of a field inherited
+ * field that has been set up at /admin/structure/field_inheritance, and
+ * the helper will find all eventinstances and make sure the new field is
+ * linked together with the relevant eventseries.
+ */
+function _dpl_update_update_field_inheritance(string $field_inheritance_name): string {
+  $ids =
+    \Drupal::entityQuery('eventinstance')
+      ->accessCheck(FALSE)
+      ->execute();
+
+  if (empty($ids) || is_int($ids)) {
+    return 'No entities to update.';
+  }
+
+  $entities =
+    \Drupal::entityTypeManager()->getStorage('eventinstance')->loadMultiple($ids);
+
+  $count = 0;
+
+  foreach ($entities as $entity) {
+    try {
+      if (!($entity instanceof EventInstance)) {
+        continue;
+      }
+
+      $event_series = $entity->getEventSeries();
+
+      if (!($event_series instanceof EventSeries)) {
+        continue;
+      }
```

Comment:
Consider dropping the two instanceof checks.

As I see it they should not be necessary due to the typing. If they actually occur they will be swallowed silently which is not good either. We are not likely to notice the total count being lower than excepted.

If the values for whatever reason not are of the excepted type and subsequent code will error and you want to proceed with your loop you could change the catch statement below to use [`\Throwable`](https://www.php.net/manual/en/class.throwable.php) instead of `\Exception`. This interface includes internal PHP errors.